Ej / Migrating project view

lightlii opened the pull request:
closes https://github.com/digidem/mapeo-desktop-shell/issues/8

## Preview
![image](https://user-images.githubusercontent.com/32508751/231175276-5283cb64-bc38-434b-9c17-f766c0524a22.png)

![image](https://user-images.githubusercontent.com/32508751/231175306-d26506a3-7fd4-4440-acb3-4f4b6de56f59.png)

lightlii committed:
fix: replace reach-router with react-router

lightlii committed:
feat: add navigation screen

lightlii committed:
feat: more work on 'navigation' view

lightlii committed:
chore: move Logo component to components directory, add Mapeo5 Logo

lightlii committed:
feat: add Migration view

lightlii committed:
chore: cleanup

lightlii committed:
chore: remove mapeo5 logo once migration complete

ErikSin commented on src/renderer/src/views/Index/index.tsx:
`index/index.tsx` is very meta 😛

ErikSin commented on src/renderer/src/views/MigratingProject/index.tsx:
```diff
@@ -0,0 +1,175 @@
+  const total = observationsTotal + mediaTotal
+  const [observationsMigrated, setObservationsMigrated] = useState(0)
+  const [mediaMigrated, setMediaMigrated] = useState(0)
+  const [progress, setProgress] = useState(0)
+  let interval: NodeJS.Timer
+
+  const getProgress = () => ((mediaMigrated + observationsMigrated) / total) * 100
+
+  useEffect(() => {
+    interval = setInterval(updateProgress, INTERVAL_DURATION)
+
+    return () => {
+      clearInterval(interval)
+    }
+  }, [])
+
+  useEffect(() => {
+    setProgress(getProgress())
+
+    console.log({ progress })
+
+    if (observationsMigrated === observationsTotal && mediaMigrated === mediaTotal) {
+      clearInterval(interval)
+    }
+  }, [observationsMigrated, observationsTotal, mediaMigrated, mediaTotal])
+
+  const updateProgress = () => {
+    const obsMigrating = Math.floor(Math.random() * 6)
+    const mediaMigrating = Math.floor(Math.random() * 6)
+
+    setObservationsMigrated((prevObservationsMigrated) =>
+      prevObservationsMigrated + obsMigrating < observationsTotal
+        ? prevObservationsMigrated + obsMigrating
+        : observationsTotal,
+    )
+    setMediaMigrated((prevMediaMigrated) =>
+      prevMediaMigrated + mediaMigrating < mediaTotal ? prevMediaMigrated + mediaMigrating : mediaTotal,
+    )
+  }
+
```
This can be simplified by just deriving `progress`:

```ts

  const updateProgress = useCallback(() => {
    const obsMigrating = Math.floor(Math.random() * 6)
    const mediaMigrating = Math.floor(Math.random() * 6)

    setObservationsMigrated((prevObservationsMigrated) =>
      prevObservationsMigrated + obsMigrating < observationsTotal
        ? prevObservationsMigrated + obsMigrating
        : observationsTotal,
    )
    setMediaMigrated((prevMediaMigrated) =>
      prevMediaMigrated + mediaMigrating < mediaTotal ? prevMediaMigrated + mediaMigrating : mediaTotal,
    )
  },[])

  useEffect(() => {
    const interval = setInterval(updateProgress, INTERVAL_DURATION)

    return () => {
      clearInterval(interval)
    }
  }, [updateProgress])

  const progress = ((mediaMigrated + observationsMigrated) / total) * 100
```

ErikSin replied:
For Reference: https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state

lightlii replied:
ah yeah, seems obvious in retrospect. Thanks!

ErikSin commented on src/renderer/src/views/Migration/index.tsx:
```diff
@@ -19,7 +19,7 @@ export const MigrationView = () => {
   const intl = useIntl()
   const theme = useTheme()
   const observations = useRef(Math.floor(Math.random() * 100))
-  const images = useRef(Math.floor(Math.random() * 100))
+  const media = useRef(Math.floor(Math.random() * 100))
```
I don't think these need to be refs, and can just be plain old `consts`.

ErikSin reviewed (approved):
looks good, just 2 minor fixes and can be merged. Thanks for the small PR :)

lightlii committed:
fix: replace reach router with react-router (#34)

* fix: replace reach router with react-router

* chore: remove redundant react-router types install

lightlii committed:
Ej/skip migration modal (#11)

* feat: add react-intl and setup basic internationalisation

* chore: add nvmrc

* refactor: extract IntlProvider and lang switcher logic into own component

* fix: import IntlSwitch properly, better messages example id

* fix: run extract-messages to update IDs in generated messages file

* fix: remove duplication in gitignore

* chore: run extract-messages pre-commit

* chore: reverse typeing translations

* feat: add reach router + basic router component

* chore: remove template 'Versions' component

* chore: cleaning

* chore: add mui

* feat: add lang switcher, default layout

* feat: add Language switcher

* feat: add theme.ts for keeping colours etc centralised

* chore: add icons-material package

* Ej/Scaffold app + internationalisation (#4)

* chore: add extract-messages script

* fix: fix app title in html file 😅

* feat: add react-intl and setup basic internationalisation

* chore: add nvmrc

* refactor: extract IntlProvider and lang switcher logic into own component

* fix: import IntlSwitch properly, better messages example id

* fix: run extract-messages to update IDs in generated messages file

* fix: remove duplication in gitignore

* chore: run extract-messages pre-commit

* chore: reverse typeing translations

* chore: remove yarn.lock

* chore: gitignore yarn.lock, generate package-lock.json

* chore: add compile script to generate translations files

* chore: add quotes on compile script

* chore: update compile script / move translations

* chore: update IntlProvider component to filter from available languages

* chore: set default lang to 'es'

* feat: add react-intl and setup basic internationalisation

* refactor: extract IntlProvider and lang switcher logic into own component

* feat: add lang switcher, default layout

* feat: add language select component

* feat: add simple splash screen

* feat: add fonts + fontface rules

* feat: add theme and theme provider

* feat: add initial migration view

* chore: add variables as constants

* chore: further styling to init screen

* feat: add skip-migration modal

* chore: add auto-generated translations

* chore: set modal size to fill most of screen

* chore: new modal complete state layout

* chore: add views to 'extract-messages' script

* Squashed commit of the following:

commit 132b8164ba8005d4b97897e42206ad546c0662b4
Author: lightlii <32508751+lightlii@users.noreply.github.com>
Date:   Wed Apr 12 12:09:48 2023 +0200

    Ej/initial screen layout (#10)

    * feat: add react-intl and setup basic internationalisation

    * refactor: extract IntlProvider and lang switcher logic into own component

    * chore: cleaning

    * Ej/Scaffold app + internationalisation (#4)

    * chore: add extract-messages script

    * fix: fix app title in html file 😅

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * chore: remove yarn.lock

    * chore: gitignore yarn.lock, generate package-lock.json

    * chore: add compile script to generate translations files

    * chore: add quotes on compile script

    * chore: update compile script / move translations

    * chore: update IntlProvider component to filter from available languages

    * chore: set default lang to 'es'

    * feat: add lang switcher, default layout

    * feat: add language select component

    * feat: add simple splash screen

    * feat: add fonts + fontface rules

    * feat: add theme and theme provider

    * feat: add initial migration view

    * chore: add variables as constants

    * chore: further styling to init screen

    * chore: remove fonts

    * Ej/Scaffold app + internationalisation (#4)

    * chore: add extract-messages script

    * fix: fix app title in html file 😅

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * chore: remove yarn.lock

    * chore: gitignore yarn.lock, generate package-lock.json

    * chore: add compile script to generate translations files

    * chore: add quotes on compile script

    * chore: update compile script / move translations

    * chore: update IntlProvider component to filter from available languages

    * chore: set default lang to 'es'

    * Ej/More setup (#9)

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * feat: add reach router + basic router component

    * chore: remove template 'Versions' component

    * chore: cleaning

    * chore: add mui

    * feat: add lang switcher, default layout

    * feat: add Language switcher

    * feat: add theme.ts for keeping colours etc centralised

    * chore: add icons-material package

    * Ej/Scaffold app + internationalisation (#4)

    * chore: add extract-messages script

    * fix: fix app title in html file 😅

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * chore: remove yarn.lock

    * chore: gitignore yarn.lock, generate package-lock.json

    * chore: add compile script to generate translations files

    * chore: add quotes on compile script

    * chore: update compile script / move translations

    * chore: update IntlProvider component to filter from available languages

    * chore: set default lang to 'es'

    * feat: add react-intl and setup basic internationalisation

    * refactor: extract IntlProvider and lang switcher logic into own component

    * feat: add lang switcher, default layout

    * feat: add language select component

    * feat: add simple splash screen

    * chore: remove unused imports

    * chore: remove redundant react fragment

    * chore: set global CSS colours from theme variables

    * chore: clean up intl provier tertiary

    * chore: remove timer hook

    ---------

    Co-authored-by: ErikSin <67773827+ErikSin@users.noreply.github.com>

    * feat: add react-intl and setup basic internationalisation

    * Ej/Scaffold app + internationalisation (#4)

    * chore: add extract-messages script

    * fix: fix app title in html file 😅

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * chore: remove yarn.lock

    * chore: gitignore yarn.lock, generate package-lock.json

    * chore: add compile script to generate translations files

    * chore: add quotes on compile script

    * chore: update compile script / move translations

    * chore: update IntlProvider component to filter from available languages

    * chore: set default lang to 'es'

    * refactor: extract IntlProvider and lang switcher logic into own component

    * feat: add lang switcher, default layout

    * feat: add fonts + fontface rules

    * feat: add theme and theme provider

    * feat: add initial migration view

    * chore: rebase fixes

    * fix: rename themeVarient => langBackgroundVarient for clarity

    * chore: remove all fonts again (?? rebase reinstated them 😑)

    * chore: remove button component

    * fix: remove hardcoded css again

    * chore: add fonts (#19)

    * fix: import useTheme (missing from rebase)

    * fix: remove fonts again (help)

    ---------

    Co-authored-by: ErikSin <67773827+ErikSin@users.noreply.github.com>

commit 0a546ca7c0fc87e7d5928ecf20cab74f419b43e8
Author: lightlii <32508751+lightlii@users.noreply.github.com>
Date:   Wed Apr 5 11:39:18 2023 +0200

    Ej/More setup (#9)

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * feat: add reach router + basic router component

    * chore: remove template 'Versions' component

    * chore: cleaning

    * chore: add mui

    * feat: add lang switcher, default layout

    * feat: add Language switcher

    * feat: add theme.ts for keeping colours etc centralised

    * chore: add icons-material package

    * Ej/Scaffold app + internationalisation (#4)

    * chore: add extract-messages script

    * fix: fix app title in html file 😅

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * chore: remove yarn.lock

    * chore: gitignore yarn.lock, generate package-lock.json

    * chore: add compile script to generate translations files

    * chore: add quotes on compile script

    * chore: update compile script / move translations

    * chore: update IntlProvider component to filter from available languages

    * chore: set default lang to 'es'

    * feat: add react-intl and setup basic internationalisation

    * refactor: extract IntlProvider and lang switcher logic into own component

    * feat: add lang switcher, default layout

    * feat: add language select component

    * feat: add simple splash screen

    * chore: remove unused imports

    * chore: remove redundant react fragment

    * chore: set global CSS colours from theme variables

    * chore: clean up intl provier tertiary

    * chore: remove timer hook

    ---------

    Co-authored-by: ErikSin <67773827+ErikSin@users.noreply.github.com>

commit 1460c04e186edeab5f24c7a84e98b262bc22895c
Merge: aa5b370 2c13648
Author: lightlii <32508751+lightlii@users.noreply.github.com>
Date:   Thu Mar 30 17:50:32 2023 +0200

    Merge branch 'develop' of github.com:digidem/mapeo-desktop-shell into develop

commit 2c13648809961197c436c55991761405ce20b346
Author: lightlii <32508751+lightlii@users.noreply.github.com>
Date:   Thu Mar 30 17:40:05 2023 +0200

    Ej/Scaffold app + internationalisation (#4)

    * chore: add extract-messages script

    * fix: fix app title in html file 😅

    * feat: add react-intl and setup basic internationalisation

    * chore: add nvmrc

    * refactor: extract IntlProvider and lang switcher logic into own component

    * fix: import IntlSwitch properly, better messages example id

    * fix: run extract-messages to update IDs in generated messages file

    * fix: remove duplication in gitignore

    * chore: run extract-messages pre-commit

    * chore: reverse typeing translations

    * chore: remove yarn.lock

    * chore: gitignore yarn.lock, generate package-lock.json

    * chore: add compile script to generate translations files

    * chore: add quotes on compile script

    * chore: update compile script / move translations

    * chore: update IntlProvider component to filter from available languages

    * chore: set default lang to 'es'

* fix: remove fonts

* chore: remove redundant useEffect and clear state on modal unmoutn

* chore: simplified button component (removed loading + disabled states)

lightlii committed:
chore: remove redundant useState for progress variable

lightlii committed:
chore: remove redundant useRef for `oberservations` and `media` variables

lightlii committed:
chore: remove fonts

lightlii committed:
Merge branch 'develop' into ej/migrating-project-view

lightlii committed:
Merge branch 'ej/skip-migration-modal' into ej/migrating-project-view
